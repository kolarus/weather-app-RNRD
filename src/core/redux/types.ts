import {Weather} from 'src/shared/api/weather/types';

export interface RootState {
  user: {
    isAuthorized: boolean;
    ui: {
      isAuthorizationInProgress: boolean;
      hasInvalidLoginAttempt: boolean;
    };
    settings: {
      units: string;
      showWeatherFor: number;
    };
  };
  weather: {
    ui: {
      isFetching: boolean;
    };
    weatherData: Nullable<Weather>;
    selectedCity: string;
    selectedCountry: string;
    lastUpdated: string;
  };
  cities: {
    ui: {
      isFetching: boolean;
    };
    cities: Array<Weather>;
  };
}

export interface Action {
  type: string;
  payload: any;
}
