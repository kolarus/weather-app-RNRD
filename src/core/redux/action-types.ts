export const USER_ACTIONS = {
  SET_HAS_INVALID_LOGIN_ATTEMPT: 'SET_HAS_INVALID_LOGIN_ATTEMPT',
  SET_IS_AUTHORIZATION_IN_PROGRESS: 'SET_IS_AUTHORIZATION_IN_PROGRESS',
  SET_IS_AUTHORIZED: 'SET_IS_AUTHORIZED',
  REQUEST_SIGN_IN: 'REQUEST_SIGN_IN',
};
