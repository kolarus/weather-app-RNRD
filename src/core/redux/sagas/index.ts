import {all} from 'redux-saga/effects';

import watchAuthorizeUser from './watch-authorize-user';
import watchFetchWeather from './watch-fetch-weather';
import watchFetchCities from './watch-fetch-cities';

export default function* rootSaga() {
  yield all([watchAuthorizeUser(), watchFetchWeather(), watchFetchCities()]);
}
