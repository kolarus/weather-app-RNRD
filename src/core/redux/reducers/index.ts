import {combineReducers} from 'redux';

import user from './user';
import weather from './weather';
import cities from './cities';

export default combineReducers({user, weather, cities});
