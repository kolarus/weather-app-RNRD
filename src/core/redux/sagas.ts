import {Alert} from 'react-native';
import {put, takeLatest, all} from 'redux-saga/effects';
import signIn from 'src/shared/api/user/sign-in';

import {
  setIsUserAuthorized,
  setIsAuthorizationInProgress,
  setHasInvalidLoginAttempt,
} from './actions/user';
import {USER_ACTIONS} from './action-types';
import {RequestSignInAction} from './actions/user/types';

export function* authorizeUser(action: RequestSignInAction) {
  try {
    yield put(setIsAuthorizationInProgress(true));
    yield signIn(action.payload.login, action.payload.password);
    yield put(setIsAuthorizationInProgress(false));
    yield put(setIsUserAuthorized(true));
  } catch (error) {
    yield put(setIsAuthorizationInProgress(false));
    yield put(setHasInvalidLoginAttempt(true));
    Alert.alert('Login Error', 'Wrong credentials', [{text: 'OK', style: 'cancel'}]);
  }
}

export function* watchAuthorizeUser() {
  yield takeLatest(USER_ACTIONS.REQUEST_SIGN_IN, authorizeUser);
}

export default function* rootSaga() {
  yield all([watchAuthorizeUser()]);
}
