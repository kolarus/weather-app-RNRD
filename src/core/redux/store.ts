import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {persistStore, persistReducer} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';

import rootSaga from './sagas';
import rootReducer from './reducers';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

const middlewares = [sagaMiddleware];

if (process.env.NODE_ENV === 'development') {
  const {logger} = require('redux-logger');

  middlewares.push(logger);
}

const persistedReducer = persistReducer(
  {
    key: 'root',
    storage: AsyncStorage,
  },
  rootReducer,
);

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(...middlewares)),
);

// @ts-ignore https://github.com/rt2zz/redux-persist/issues/1140
export const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);
