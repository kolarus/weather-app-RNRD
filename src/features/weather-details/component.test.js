import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {applyMiddleware, createStore} from 'redux';
import createSagaMiddleware from 'redux-saga';

import WeatherDetails from './component';
import rootSaga from '../../core/redux/sagas';

let store = createSagaMiddleware();

beforeEach(() => {
  const sagaMiddleware = createSagaMiddleware();
  store = createStore((state) => state, initialState, applyMiddleware(sagaMiddleware));
  sagaMiddleware.run(rootSaga);
  jest.useFakeTimers();
  fetch.resetMocks();
});

jest.mock('@react-navigation/stack', () => ({
  useHeaderHeight: () => 100,
}));

jest.mock('@react-navigation/core', () => ({
  useRoute: () => ({}),
}));

const initialState = {
  weather: {
    ui: {
      isFetching: false,
    },
    weatherData: {
      cod: '200',
      message: 0,
      cnt: 40,
      list: [
        {
          dt: 1606510800,
          main: {
            temp: 4.53,
            feels_like: 0.81,
            temp_min: 4.03,
            temp_max: 4.53,
            pressure: 1018,
            sea_level: 1018,
            grnd_level: 1000,
            humidity: 88,
            temp_kf: 0.5,
          },
          weather: [
            {
              id: 804,
              main: 'Clouds',
              description: 'overcast clouds',
              icon: '04n',
            },
          ],
          clouds: {
            all: 87,
          },
          wind: {
            speed: 3.1,
            deg: 216,
          },
          visibility: 10000,
          pop: 0,
          sys: {
            pod: 'n',
          },
          dt_txt: '2020-11-27 21:00:00',
        },
        {
          dt: 1606521600,
          main: {
            temp: 3.38,
            feels_like: 0.11,
            temp_min: 2.93,
            temp_max: 3.38,
            pressure: 1019,
            sea_level: 1019,
            grnd_level: 1000,
            humidity: 91,
            temp_kf: 0.45,
          },
          weather: [
            {
              id: 804,
              main: 'Clouds',
              description: 'overcast clouds',
              icon: '04n',
            },
          ],
          clouds: {
            all: 95,
          },
          wind: {
            speed: 2.29,
            deg: 221,
          },
          visibility: 10000,
          pop: 0,
          sys: {
            pod: 'n',
          },
          dt_txt: '2020-11-28 00:00:00',
        },
        {
          dt: 1606532400,
          main: {
            temp: 2.63,
            feels_like: -0.53,
            temp_min: 2.47,
            temp_max: 2.63,
            pressure: 1020,
            sea_level: 1020,
            grnd_level: 1001,
            humidity: 87,
            temp_kf: 0.16,
          },
          weather: [
            {
              id: 804,
              main: 'Clouds',
              description: 'overcast clouds',
              icon: '04n',
            },
          ],
          clouds: {
            all: 98,
          },
          wind: {
            speed: 1.82,
            deg: 219,
          },
          visibility: 10000,
          pop: 0,
          sys: {
            pod: 'n',
          },
          dt_txt: '2020-11-28 03:00:00',
        },
      ],
      city: {
        id: 706483,
        name: 'Kharkiv',
        coord: {
          lat: 50,
          lon: 36.25,
        },
        country: 'UA',
        population: 1430885,
        timezone: 7200,
        sunrise: 1606453605,
        sunset: 1606484343,
      },
    },
    selectedCity: 'Kharkiv',
    selectedCountry: 'ua',
    lastUpdated: '0:00',
  },
  user: {
    settings: {
      units: 'metric',
      showWeatherFor: 5,
    },
  },
};

test('WeatherDetails should match snapshot', () => {
  const {toJSON} = render(
    <Provider store={createStore((state) => state, initialState)}>
      <WeatherDetails />
    </Provider>,
  );

  expect(toJSON()).toMatchSnapshot();
});

test('WeatherDetails should fetch data when refresh icon pressed', async () => {
  const {getByTestId} = render(
    <Provider store={store}>
      <WeatherDetails />
    </Provider>,
  );

  fetch.resetMocks();
  fireEvent(getByTestId('WeatherDetails.refresh'), 'press');

  await expect(fetch.mock.calls.length).toBe(1);
});
