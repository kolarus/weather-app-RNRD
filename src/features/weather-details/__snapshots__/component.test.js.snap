// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetails should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393c50",
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
      },
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#080b24",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "paddingHorizontal": "3%",
        },
        Object {
          "paddingTop": 100,
        },
      ],
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "bottom": 0,
        "justifyContent": "flex-start",
        "left": 0,
        "opacity": 0.1,
        "position": "absolute",
        "resizeMode": "cover",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
  <RNCSafeAreaView>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "PT Sans",
              "fontSize": 15,
            },
            Object {
              "fontFamily": "PT Sans Narrow",
              "fontSize": 20,
            },
            Object {},
          ]
        }
      >
        0:00
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "1deg",
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="WeatherDetails.refresh"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "PT Sans",
              "fontSize": 15,
            },
            Object {
              "fontFamily": "PT Sans Narrow",
              "fontSize": 50,
              "letterSpacing": 10,
              "marginLeft": -5,
              "textTransform": "uppercase",
            },
            Object {},
          ]
        }
      >
        Kharkiv
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "PT Sans",
              "fontSize": 15,
            },
            Object {
              "fontFamily": "PT Sans Narrow",
              "fontSize": 25,
              "letterSpacing": 5,
              "marginLeft": -2,
              "textTransform": "uppercase",
            },
            Object {},
          ]
        }
      >
        Friday, November, 27
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "marginTop": 50,
          "paddingBottom": 50,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 2,
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 80,
              },
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 3,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontFamily": "PT Sans",
                  "fontSize": 15,
                },
                Array [
                  Object {
                    "fontFamily": "PT Sans Narrow",
                  },
                  Object {
                    "fontSize": 60,
                  },
                ],
                Object {},
              ]
            }
          >
            5
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontFamily": "PT Sans",
                  "fontSize": 15,
                },
                Array [
                  Object {
                    "fontSize": 36,
                    "lineHeight": 43.199999999999996,
                  },
                  Object {
                    "fontFamily": "PT Sans Narrow",
                  },
                ],
                Object {},
              ]
            }
          >
            o
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "PT Sans",
                "fontSize": 15,
              },
              Object {
                "fontFamily": "PT Sans Narrow",
                "fontSize": 13,
                "marginLeft": 10,
                "textTransform": "uppercase",
              },
              Object {},
            ]
          }
        >
          // 
          overcast clouds
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "PT Sans",
                "fontSize": 15,
              },
              Object {
                "fontFamily": "PT Sans Narrow",
                "fontSize": 20,
                "marginTop": 10,
              },
              Object {},
            ]
          }
        >
          15:39
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 120,
        }
      }
    >
      <RCTScrollView
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "PT Sans",
                    "fontSize": 15,
                  },
                  Object {
                    "fontSize": 12,
                  },
                  Object {},
                ]
              }
            >
              23:00
            </Text>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  Object {
                    "marginHorizontal": 20,
                    "marginVertical": 10,
                  },
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "PT Sans",
                      "fontSize": 15,
                    },
                    Array [
                      undefined,
                      Object {
                        "fontSize": 25,
                      },
                    ],
                    Object {},
                  ]
                }
              >
                5
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "PT Sans",
                      "fontSize": 15,
                    },
                    Array [
                      Object {
                        "fontSize": 15,
                        "lineHeight": 18,
                      },
                      undefined,
                    ],
                    Object {},
                  ]
                }
              >
                o
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "PT Sans",
                    "fontSize": 15,
                  },
                  Object {
                    "fontSize": 12,
                  },
                  Object {},
                ]
              }
            >
              2:00
            </Text>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  Object {
                    "marginHorizontal": 20,
                    "marginVertical": 10,
                  },
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "PT Sans",
                      "fontSize": 15,
                    },
                    Array [
                      undefined,
                      Object {
                        "fontSize": 25,
                      },
                    ],
                    Object {},
                  ]
                }
              >
                3
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "PT Sans",
                      "fontSize": 15,
                    },
                    Array [
                      Object {
                        "fontSize": 15,
                        "lineHeight": 18,
                      },
                      undefined,
                    ],
                    Object {},
                  ]
                }
              >
                o
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "PT Sans",
                    "fontSize": 15,
                  },
                  Object {
                    "fontSize": 12,
                  },
                  Object {},
                ]
              }
            >
              5:00
            </Text>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  Object {
                    "marginHorizontal": 20,
                    "marginVertical": 10,
                  },
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "PT Sans",
                      "fontSize": 15,
                    },
                    Array [
                      undefined,
                      Object {
                        "fontSize": 25,
                      },
                    ],
                    Object {},
                  ]
                }
              >
                3
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "PT Sans",
                      "fontSize": 15,
                    },
                    Array [
                      Object {
                        "fontSize": 15,
                        "lineHeight": 18,
                      },
                      undefined,
                    ],
                    Object {},
                  ]
                }
              >
                o
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "padding": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 30,
              },
              Object {
                "marginHorizontal": 20,
                "marginVertical": 10,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "PT Sans",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontSize": 25,
              },
            ]
          }
        >
          88
          %
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "padding": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 30,
              },
              Object {
                "marginHorizontal": 20,
                "marginVertical": 10,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontFamily": "PT Sans",
                  "fontSize": 15,
                },
                Array [
                  undefined,
                  Object {
                    "fontSize": 25,
                  },
                ],
                Object {},
              ]
            }
          >
            3.1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontFamily": "PT Sans",
                  "fontSize": 15,
                },
                Array [
                  Object {
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                  undefined,
                ],
                Object {},
              ]
            }
          >
            m/c
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "padding": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 30,
              },
              Object {
                "marginHorizontal": 20,
                "marginVertical": 10,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "PT Sans",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontSize": 25,
              },
            ]
          }
        >
          7:06
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "padding": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 30,
              },
              Object {
                "marginHorizontal": 20,
                "marginVertical": 10,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "PT Sans",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontSize": 25,
              },
            ]
          }
        >
          15:39
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
